exercise/api/custom_serializers.py

```diff
@@ -49,10 +49,20 @@ def submission_url(submission_id):
                 'submission_id': submission_id
             }, request=request)
 
+        def submission_obj(submmission_cached):
+            id_ = submmission_cached['id']
+            return {
+                'id': id_,
+                'url': submission_url(id_),
+                'submission_time': submmission_cached['date'],
+                'grade': submmission_cached['points'], # NOTE: model has this in field `grade`, TODO rename
+            }
+
         exercise_data = {
             'url': exercise_url(entry['id']),
             'best_submission': submission_url(entry['best_submission']),
             'submissions': [submission_url(s['id']) for s in entry['submissions']],
+            'submissions_and_points': [submission_obj(s) for s in entry['submissions']],
```

Comment:
We need to consider the word used in the output. `submissions_and_points` is not the best one necessarily. Since the word `grade` is used in the object, should this be `submissions_with_grades`?

Reply:
Yes `submissions_with_grades` could be good if we use the key grade. makes sense to use the same term here.

Reply:
Depends if we want to prefer word `points` or `grade` for this purpose. Details only has `grade`, because that is the name in the model. I should have used `points`, but I didn't understand enough at the time.

Reply:
Maybe `submissions_with_points` so that it is ready for the future if `grade` is changed to `points` some day.

Reply:
:+1: `with` is better than `and`.